lib/tx_build/preconditions.ex

```diff
@@ -0,0 +1,185 @@
+defmodule Stellar.TxBuild.Preconditions do
+  @moduledoc """
+  `Preconditions` struct definition.
+  """
+  alias Stellar.TxBuild.{
+    TimeBounds,
+    LedgerBounds,
+    SequenceNumber,
+    SignerKey
+  }
+
+  alias StellarBase.XDR.TimeBounds, as: TimeBoundsXDR
+
+  alias StellarBase.XDR.{
+    Preconditions,
+    PreconditionsV2,
+    PreconditionType,
+    Duration,
+    OptionalSequenceNumber,
+    SignerKeyList,
+    TimePoint,
+    UInt32,
+    Void
+  }
+
+  @behaviour Stellar.TxBuild.XDR
+
+  @type precond_v2 :: %{
+          time_bounds: TimeBounds.t(),
+          ledger_bounds: LedgerBounds.t(),
+          min_seq_num: SequenceNumber.t(),
+          min_seq_age: non_neg_integer(),
+          min_seq_ledger_gap: non_neg_integer(),
+          extra_signers: list(SignerKey.t())
+        }
+  @type validation ::
+          {:ok, non_neg_integer()} | {:ok, struct()} | {:ok, list(struct())} | {:error, atom()}
+  @type type :: :precond_v2 | :precond_time | :none
+  @type preconditions :: precond_v2() | TimeBounds.t() | nil
+
+  @type t :: %__MODULE__{type: type(), preconditions: preconditions()}
+
+  defstruct [:type, :preconditions]
+
+  @impl true
+  def new(preconditions \\ [], opts \\ [])
+
+  def new([], _opts) do
+    %__MODULE__{type: :none, preconditions: nil}
+  end
+
+  def new([time_bounds: %TimeBounds{} = time_bounds], _opts) do
+    %__MODULE__{type: :precond_time, preconditions: time_bounds}
+  end
+
+  def new(preconditions, _opts) do
```

Comment:
@CristhianRodriguezMolina @FelipeGuzmanSierra How to initialize preconditions of type `: precond_time` or `:none`

Reply:
1. Type `:none`, when we create the preconditions using `Preconditions.new()` or `Preconditions.new([])`.
2. Type `:precond_time`, when we create the preconditions only with time_bounds like `Preconditions.new([time_bounds: TimeBounds.new()])`